resources/packs/gcp/kms.go

```diff
@@ -0,0 +1,372 @@
+	}
+	return name, nil
+}
+
+func (g *mqlGcpProjectKmsServiceKeyringCryptokeyVersionAttestation) id() (string, error) {
+	name, err := g.CryptoKeyVersionName()
+	if err != nil {
+		return "", err
+	}
+	return fmt.Sprintf("%s/attestation", name), nil
+}
+
+func (g *mqlGcpProjectKmsServiceKeyringCryptokeyVersionExternalProtectionLevelOptions) id() (string, error) {
+	name, err := g.CryptoKeyVersionName()
+	if err != nil {
+		return "", err
+	}
+	return fmt.Sprintf("%s/externalProtectionLevelOptions", name), nil
+}
+
+func (g *mqlGcpProjectKmsServiceKeyringCryptokeyVersionAttestationCertificatechains) id() (string, error) {
+	name, err := g.CryptoKeyVersionName()
+	if err != nil {
+		return "", err
+	}
+	return fmt.Sprintf("%s/attestation/certchains", name), nil
+}
+
+func (g *mqlGcpProjectKmsService) GetLocations() ([]interface{}, error) {
+	projectId, err := g.ProjectId()
+	if err != nil {
+		return nil, err
+	}
+
+	provider, err := gcpProvider(g.MotorRuntime.Motor.Provider)
+	if err != nil {
+		return nil, err
+	}
+
+	creds, err := provider.Credentials(kms.DefaultAuthScopes()...)
+	if err != nil {
+		return nil, err
+	}
+
+	ctx := context.Background()
+
+	kmsSvc, err := kms.NewKeyManagementClient(ctx, option.WithCredentials(creds))
+	if err != nil {
+		return nil, err
+	}
+	defer kmsSvc.Close()
+
+	var locations []interface{}
+	it := kmsSvc.ListLocations(ctx, &location.ListLocationsRequest{Name: fmt.Sprintf("projects/%s", projectId)})
+	for {
+		l, err := it.Next()
+		if err == iterator.Done {
+			break
+		}
+		if err != nil {
+			return nil, err
+		}
+
+		locations = append(locations, l.LocationId)
+	}
+	return locations, nil
+}
+
+func (g *mqlGcpProjectKmsService) GetKeyrings() ([]interface{}, error) {
+	projectId, err := g.ProjectId()
+	if err != nil {
+		return nil, err
+	}
+
+	locations, err := g.Locations()
+	if err != nil {
+		return nil, err
+	}
+
+	provider, err := gcpProvider(g.MotorRuntime.Motor.Provider)
+	if err != nil {
+		return nil, err
+	}
+
+	creds, err := provider.Credentials(kms.DefaultAuthScopes()...)
+	if err != nil {
+		return nil, err
+	}
+
+	ctx := context.Background()
+
+	kmsSvc, err := kms.NewKeyManagementClient(ctx, option.WithCredentials(creds))
+	if err != nil {
+		return nil, err
+	}
+	defer kmsSvc.Close()
+
+	var keyrings []interface{}
+	var wg sync.WaitGroup
+	wg.Add(len(locations))
+	mux := &sync.Mutex{}
+
+	for _, location := range locations {
+		go func(svc *kms.KeyManagementClient, project string, location string) {
+			defer wg.Done()
+			it := kmsSvc.ListKeyRings(ctx,
+				&kmspb.ListKeyRingsRequest{Parent: fmt.Sprintf("projects/%s/locations/%s", projectId, location)})
+			for {
+				k, err := it.Next()
+				if err == iterator.Done {
+					break
+				}
+				if err != nil {
+					log.Error().Err(err)
+					return
+				}
+
+				created := k.CreateTime.AsTime()
+				mqlKeyring, err := g.MotorRuntime.CreateResource("gcp.project.kmsService.keyring",
+					"projectId", projectId,
+					"name", k.Name,
```

Comment:
This name should be just the key ring name:

```
gcp.project.kms.keyrings: [
  0: {
    created: 2022-12-19 15:17:46.974842182 +0000 UTC
    projectId: "example-test"
    cryptokeys: [
  0: gcp.project.kmsService.keyring.cryptokey name="projects/example-test/locations/global/keyRings/hosted-scanner/cryptoKeys/hosted-scanner-key" purpose="ENCRYPT_DECRYPT"
    ]
    name: "projects/example-test/locations/global/keyRings/hosted-scanner"
    location: "global"
  }
]
```

It would be more helpful for the user to have the information available as a parsed info. The same applies to `cryptokey`.